router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go

```diff
@@ -208,9 +208,19 @@ func (m *MarketoAPIService) GetLeadStatus(url string) ([]map[string]string, *API
 
 	reader := csv.NewReader(strings.NewReader(string(body)))
 
-	rows, err := reader.ReadAll()
-	if err != nil {
-		return nil, &APIError{StatusCode: 500, Category: "Retryable", Message: "Error in parsing csv response"}
+	// read each row one by one
+	rows := make([][]string, 0)
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
```

Comment:
looks even error comes, record is returned, so can use it, can you try this by sample code

Reply:
we should handle this by removing \n characters in transformer

Reply:
Sure, will take it up